core/server/proxy/src/main/java/alluxio/proxy/s3/MultipartUploadCleaner.java

```diff
@@ -0,0 +1,288 @@
+        long lastModificationTimeMs = status.getLastModificationTimeMs();
+        delay = lastModificationTimeMs + mTimeout - curTime;
+        if (delay <= 0) {
+          // check object, when merge multipart upload, it may be timeout
+          try {
+            AlluxioURI uri = new AlluxioURI(objectPath);
+            status = mFs.getStatus(uri);
+            lastModificationTimeMs = status.getLastModificationTimeMs();
+            delay = lastModificationTimeMs + mTimeout - curTime;
+            if (delay <= 0) {
+              mFs.delete(tmpUri, DeletePOptions.newBuilder().setRecursive(true).build());
+              LOG.info("Abort multipart upload {} in bucket {} with uploadId {}.",
+                  object, bucket, uploadId);
+            }
+          } catch (FileDoesNotExistException e) {
+            mFs.delete(tmpUri, DeletePOptions.newBuilder().setRecursive(true).build());
+            LOG.info("Abort multipart upload {} in bucket {} with uploadId {}.",
+                object, bucket, uploadId);
+          }
+        }
+      }
+    } catch (FileDoesNotExistException ignored) {
+      return delay;
+    }
+    return delay;
+  }
+
+  /**
+   * Get multipart uploadId.
+   *
+   * @param bucket the bucket name
+   * @param object the object name
+   */
+  private Long getMultipartUploadId(String bucket, String object)
+      throws IOException, AlluxioException {
+    final String bucketPath = AlluxioURI.SEPARATOR  + bucket;
+    String multipartTemporaryDir =
+        S3RestUtils.getMultipartTemporaryDirForObject(bucketPath, object);
+    try {
+      URIStatus status = mFs.getStatus(new AlluxioURI(multipartTemporaryDir));
+      return status.getFileId();
+    } catch (FileDoesNotExistException | InvalidPathException e) {
+      return null;
+    }
+  }
+
+  /**
+   * Abort Multipart upload task.
+   */
+  public class AbortTask implements Runnable {
+    private final String mBucket;
+    private final String mObject;
+    private final Long mUploadId;
+    private int mRetryCount;
+
+    /**
+     * Creates a new instance of {@link AbortTask}.
+     *
+     * @param bucket the bucket name
+     * @param object the object name
+     * @param uploadId multipart upload tmp directory fileId
+     */
+    public AbortTask(String bucket, String object,
+                     Long uploadId) {
+      mBucket = bucket;
+      mObject = object;
+      mUploadId = uploadId;
+      mRetryCount = 0;
+    }
+
+    @Override
+    public void run() {
+      try {
+        long delay = tryAbortMultipartUpload(mBucket, mObject, mUploadId);
+        if (delay > 0) {
+          apply(this, delay);
+        } else {
+          removeTaskRecord(this);
+        }
+      } catch (IOException | AlluxioException e) {
+        mRetryCount++;
+        LOG.error("Failed abort multipart upload {} in bucket {} with uploadId {} with error {}"
+            + " after retry {} count.", mObject, mBucket, mUploadId, e, mRetryCount);
+        e.printStackTrace();
```

Comment:
we should not print stack traces directly to the stdout. Instead I suggest logging with the following format:

```java
 LOG.error("Failed abort multipart upload {} in bucket {} with uploadId {} after {} retries with error {}"
            + " after retry {} count.", mObject, mBucket, mUploadId, retryCount, e);
```

Reply:
Do you means this?
```
LOG.error("Failed abort multipart upload {} in bucket {} with uploadId {} after {} retries with error {}.", 
            mObject, mBucket, mUploadId, mRetryCount, e);
```